### PR TITLE
suite-sparse: migrate from homebrew/science

### PR DESCRIPTION
**Pull request by fxcoudert**
See https://github.com/Homebrew/homebrew-core/issues/15619

### CODE REVIEW

File: Formula/suite-sparse.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                  "TBB=-L#{Formula["tbb"].opt_lib} -ltbb",
+                  "BLAS=-framework Accelerate",
+                  "LAPACK=$(BLAS)",
+                  "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                  "MY_METIS_INC=#{Formula["metis"].opt_include}" ]
+
+    # Only building libraries
+    system "make", "library", *make_args
+    system "make", "install", "INSTALL=#{prefix}", *make_args
+
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: This should probably be the end of the install method.

**Comment by fxcoudert** [comment]
Updated to version 4.5.5. Removed `tbb` which is a mess to configure correctly, and is only used in one sublibrary. Do not build/install demos, but copy over a simple test.

File: Formula/suite-sparse.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class SuiteSparse < Formula
+  desc "Suite of Sparse Matrix Software"
+  homepage "http://faculty.cse.tamu.edu/davis/suitesparse.html"
+  url "http://faculty.cse.tamu.edu/davis/SuiteSparse/SuiteSparse-4.5.5.tar.gz"
+  sha256 "b9a98de0ddafe7659adffad8a58ca3911c1afa8b509355e7aa58b02feb35d9b6"
+
+  # SuiteSparse ships with metis5 but we use the Homebrew version
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: comment isn't needed

File: Formula/suite-sparse.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "metis"
+
+  def install
+    make_args = [ "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: this should be inlined since nothing is optionally appended to the array

Comment (fxcoudert) [inline Formula/suite-sparse.rb]: We're using it twice

File: Formula/suite-sparse.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                  "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                  "MY_METIS_INC=#{Formula["metis"].opt_include}" ]
+
+    # Only building libraries
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: comment not needed

File: Formula/suite-sparse.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: does make library install not work?

File: Formula/suite-sparse.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  depends_on "metis"
+
+  def install
+    system "make", "library", "install", "INSTALL=#{prefix}",
+                   "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
+                   "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                   "MY_METIS_INC=#{Formula["metis"].opt_include}"
+
+    # Install a simple test
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: The comment probably isn't needed at all, but if you do want to keep it, it'll fit inline.
```
    pkgshare.install "KLU/Demo/klu_simple.c" # install a simple test 
```

File: Formula/suite-sparse.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    pkgshare.install "KLU/Demo/klu_simple.c"
+  end
+
+  test do
+    system ENV.cc, "-o", "test", pkgshare/"klu_simple.c", "-L#{lib}",
+                   "-I#{include}", "-lsuitesparseconfig", "-lklu"
+    system "./test"
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: anything we can verify here?

File: Formula/suite-sparse.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "b9a98de0ddafe7659adffad8a58ca3911c1afa8b509355e7aa58b02feb35d9b6"
+
+  depends_on "metis"
+
+  def install
+    system "make", "library", "install", "INSTALL=#{prefix}",
+                   "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
+                   "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+                   "MY_METIS_INC=#{Formula["metis"].opt_include}"
+
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: this blank line can go away

File: Formula/suite-sparse.rb (hunk 6 above)
Comment (fxcoudert) [inline Formula/suite-sparse.rb]: Not really, the output is numerical so subject to slight variations. I think compiling, linking and running is good enough.

File: Formula/suite-sparse.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    pkgshare.install "KLU/Demo/klu_simple.c"
+  end
+
+  test do
+    system ENV.cc, "-o", "test", pkgshare/"klu_simple.c", "-L#{lib}",
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: would switch the order of "-I#{include}" and "-L#{lib}", though "-I#{include}" is not even needed

File: Formula/suite-sparse.rb (hunk 7 above)
Comment (fxcoudert) [inline Formula/suite-sparse.rb]: You have too many style preferences :) you need to write a book

File: Formula/suite-sparse.rb (hunk 6 above)
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: ok

File: Formula/suite-sparse.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  sha256 "b9a98de0ddafe7659adffad8a58ca3911c1afa8b509355e7aa58b02feb35d9b6"
+
+  depends_on "metis"
+
+  def install
+    args = [
+      "INSTALL=#{prefix}",
+      "BLAS=-framework Accelerate", "LAPACK=$(BLAS)",
+      "MY_METIS_LIB=-L#{Formula["metis"].opt_lib} -lmetis",
+      "MY_METIS_INC=#{Formula["metis"].opt_include}"
```
Comment (ilovezfs) [inline Formula/suite-sparse.rb]: I think it will require a comma at the end of this line

Comment (fxcoudert) [inline Formula/suite-sparse.rb]: Nope, it works and audit is not complaining :)

Comment (ilovezfs) [inline Formula/suite-sparse.rb]: Putting `"BLAS=-framework Accelerate", "LAPACK=$(BLAS)",` on a single line defeated the check, unfortunately

Comment (ilovezfs) [inline Formula/suite-sparse.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/15913.

**Pull request closed without merging**